presentation/src/main/java/com/dcs/presentation/core/model/SettingUiState.kt

```diff
@@ -35,28 +38,36 @@ sealed interface UserProfile {
         operator fun invoke(
             user: com.dcs.domain.model.User?,
         ): UserProfile {
-            return if (user != null) {
-                User(
-                    id = user.id,
-                    tmdbId = user.tmdbId,
-                    includeAdult = user.includeAdult,
-                    name = user.name,
-                    username = user.username,
-                    sessionId = user.sessionId,
-                    avatarImageUrl = if (user.avatar.tmdb.value != null) {
-
-                        "${BuildConfig.TMDB_IMAGE_URL}/w200${user.avatar.tmdb.value}"
-                    } else {
-                        "${BuildConfig.TMDB_GRAVATR_URL}${user.avatar.gravatar.value}.jpg?s=200"
-                    },
-                )
-            } else {
-                NotLoggedIn
+            if (user == null) {
+                return NotLoggedIn
```

Comment:
Return이 왜 if 문 안쪽으로 들어왔지?

Reply:
if else 문이 지저분해서 early return 패턴 사용했어요

Reply:
분기가 복잡한 것도 아니고 결과를 예상하기 어려운 것도 아니므로 early return 패턴은 사용할 필요 없이 exit point는 하나로 해서 이전 스타일로 가는게 나은 것 같음.

Reply:
```kotlin
val user = user ?: return NotLoggedIn
            return User(
                id = user.id,
                tmdbId = user.tmdbId,
                includeAdult = user.includeAdult,
                name = user.name,
                username = user.username,
                sessionId = user.sessionId,
                avatarImageUrl = if (user.avatar.tmdb.value != null) {
                    "${BuildConfig.TMDB_IMAGE_URL}/w200${user.avatar.tmdb.value}"
                } else {
                    "${BuildConfig.TMDB_GRAVATR_URL}${user.avatar.gravatar.value}.jpg?s=200"
                },
            )
```

이건 어떠신가요?

Reply:
return user?.let {
 User(..)
} ?: NotLoggedIn
이게 나을듯?

Reply:
최대한 scope 함수 안 쓰려고 한거라..그리고 var 변수가 아닌 다른 타입의 변수에 대해서 단순 null check용으로 let을 쓰는건 권장하지 않는 걸로 알고 있어서요. [참고](https://medium.com/mobile-app-development-publication/kotlin-dont-just-use-let-7e91f544e27f)
이전 걸로 돌릴까요.......

Reply:
단순 null check로 볼게 아니라 user를 user?.let{ it.toUser() } 로 변환해서 반환값에 쓰이는 다른 객체를 넘겨준다고 봐야지.

Reply:
user?.let{ it.toUser() } 까지만 보면 그런데 뒤에 ?: NotLoggedIn이 있어서 null check까지 의미가 들어 가서 그냥 빨리 값 처리 하는게 좋지 않나 생각을 했어요. scope 열고 닫고 하는게 그렇게 눈에 잘 익은 형태가 아니라서요.

Reply:
많이 쓰는 방식임 ㅋ

Reply:
흔히들 많이 쓰긴하지만 들여쓰기랑 brace 때문에 가독성 떨어지고 early return pattern이 이런 부분에서 확실히 장점이 있어서 좋다라는 의견들을 많이 들어서요. 저도 그래서 early return pattern을 많이 사용하는 편이구요.  kotlin scope function 경우에는 연속성(chaining)을 가지거나 제한적인 상황 쓰는 편이긴합니다. 그렇지 않은 상황에서는 들여쓰기 들어가는건 똑같아서요.

아무튼 그래서 그냥 이전 코드로 돌렸습니다. scope 함수보다는 if else로 두는게 더 읽기 편하다는 생각이 들어서요.
https://github.com/Edgar-J-Shin/AllAboutMovie/pull/28/commits/9ac726f153e30da3f960a8d0a601422a8eb046ed

Reply:
```suggestion
   if (user == null) return NotLoggedIn

    return user.toUser()
    )```

early return 방식에서는 이렇게 많이 쓰지 않나?
null 체크 후에 리턴 구문 처리한 다음에 본문 처리하는 방식?

Reply:
```
if (user == null) return NotLoggedIn

or 

if (user == null) {
   return NotLoggedIn
}

// brace 쓰는건 취향차이..
```

이렇게 하고 뒷쪽 처리할 로직에 여러 방안으로 하긴 하죠 형이 작성한 것 처럼 함수로 한번 더 감싸서 저함수로 내려가서 책임 분리 하는 것도 권장되는 방법이기도 하죠. 함수로 빼는건 작년에 핫해서 사람들이 많이 있는 파이브 라인스라는 책에서 봤습니다. 

```kotlin
val user = user ?: return NotLoggedIn

return createUserProfileUser(user)
```
User 객체 선언 부분을 함수로 한번 더 빼는걸로 이렇게 할 수도 있을 것 같아요. 뭐 같은 방식이지만 ㅋㅋㅋㅋㅋ 이렇게 하니까 invoke에 또 invoke 하는 느낌이긴 하네요..

Reply:
val user = user 이부분이 이상한데? ㅋ if문으로 가는게 나을 듯.

Reply:
if문은 java 스타일,  엘비스는 코틀린 스타일이라는 말을 어디서 듣긴 했습니다ㅋㅋㅋ
일단 이전 코드로 돌려서 기존대로 가시죠

Reply:
return if ( ... ) {
 } else {
 return NotLoggedIn
}

지금 마지막 버전이 이렇게 되어있는데 잘못 본건가??

Reply:
오 return이 잘 못 들어 가 있었네요. 제거 했습니다.